<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Visualizing the geo-biodiversity profiles and the hierarchical structure of biodiversity • HierDmap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Visualizing the geo-biodiversity profiles and the hierarchical structure of biodiversity">
<meta property="og:description" content="HierDmap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">HierDmap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.5.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/index.html">UNKNOWN TITLE</a>
    </li>
    <li>
      <a href="../articles/introduction.html">Visualizing the geo-biodiversity profiles and the hierarchical structure of biodiversity</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xinghuq/HierDmap/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="introduction_files/htmlwidgets-1.5.3/htmlwidgets.js"></script><script src="introduction_files/d3-3.5.6/./d3.min.js"></script><link href="introduction_files/circlepackeR-0.1/./style.css" rel="stylesheet">
<script src="introduction_files/circlepackeR-binding-0.0.0.9000/circlepackeR.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Visualizing the geo-biodiversity profiles and the hierarchical structure of biodiversity</h1>
                        <h4 class="author">Xinghu Qin –School of Biology, University of St Andrews</h4>
            
            <h4 class="date">19-04-2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/xinghuq/HierDmap/blob/master/vignettes/introduction.Rmd"><code>vignettes/introduction.Rmd</code></a></small>
      <div class="hidden name"><code>introduction.Rmd</code></div>

    </div>

    
    
<div id="instruction" class="section level1">
<h1 class="hasAnchor">
<a href="#instruction" class="anchor"></a>Instruction</h1>
<p>This vignette provides a step-by-step tutorial for visualizing the geo-genetic diversity profiles and the hierarchical structure of genetic diversity. In this tutorial, we will use the simulated microsatellite genotypes under a hierarchical island model to visualize the genetic diversity in the hierarchy. We then use the HDGP data to show the genetic diversity-density map.</p>
</div>
<div id="diversity-profiles-and-geo-biodiversity-map" class="section level1">
<h1 class="hasAnchor">
<a href="#diversity-profiles-and-geo-biodiversity-map" class="anchor"></a>Diversity profiles and geo-biodiversity map</h1>
<p>The diversity profile, <em>q^D</em> , based on the Hill numbers (q=0, 1, 2), provides a complementary description of the variations in aggregates (Chao, A. et al, 2014; Gaggiotti, et al, 2018; Jost, L. et al, 2018; Sherwin, W. 2018). In the natural world, aggregates (populations and communities) are typically spatially-structured. Decomposing aggregates into global and local aggregates has allowed us to better understand its components in a hierarchy. For example, populations sampled from large spatial areas (i.e., continents) can be partitioned into several regions, sub-regions, populations, and sub-populations based on the nested structure of the metapopulations. Currently, packages explicitly used to visualize the hierarchical structure of diversities are still lacking. I present the “HierDmap” package here. The main function of this package is to intuitively visualize the diversity (species diversity and genetic diversity) in a hierarchy and plot the diversity profiles onto a geographic map.</p>
<p>We use two datasets to demonstrate the usefulness of this package. We first use simulated microsatellite genotypes with 16 populations nested in four regions to visualize the diversity in the hierarchy. The hierarchy of the structure can be formatted using “HierStr” function from “HierDpart” package (Qin, X, 2019). Besides, we use the Human Genome Diversity Project (HGDP) data (Cann HM,et al. 2002) to plot the genetic diversity onto a geographic map.</p>
<div id="install-and-load-the-dataset" class="section level2">
<h2 class="hasAnchor">
<a href="#install-and-load-the-dataset" class="anchor"></a>Install and load the dataset</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## or you can get the latest version of HierDpart from github</span>
<span class="co">#if (!requireNamespace("devtools", quietly = TRUE))</span>
<span class="co">#    install.packages("devtools")</span>
<span class="co">#if (!requireNamespace("circlepackeR", quietly = TRUE))</span>
<span class="co">#devtools::install_github("jeromefroe/circlepackeR")</span>

<span class="co">#if (!requireNamespace("HierDmap", quietly = TRUE))</span>
<span class="co">#devtools::install_github("xinghuq/HierDmap")</span>
<span class="co">#install_github("xinghuq/HierDmap")</span>

<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/xinghuq/HierDmap">"HierDmap"</a></span><span class="op">)</span>
<span class="co">#&gt; Registered S3 method overwritten by 'spdep':</span>
<span class="co">#&gt;   method   from</span>
<span class="co">#&gt;   plot.mst ape</span>
<span class="co">#&gt; Registered S3 method overwritten by 'GGally':</span>
<span class="co">#&gt;   method from   </span>
<span class="co">#&gt;   +.gg   ggplot2</span>
<span class="co">#&gt; Registered S3 method overwritten by 'pegas':</span>
<span class="co">#&gt;   method      from</span>
<span class="co">#&gt;   print.amova ade4</span>
<span class="co">#&gt; Warning: replacing previous import 'rgeos::union' by 'dplyr::union' when loading</span>
<span class="co">#&gt; 'HierDmap'</span>
<span class="co">#&gt; Warning: replacing previous import 'rgeos::intersect' by 'dplyr::intersect' when</span>
<span class="co">#&gt; loading 'HierDmap'</span>
<span class="co">#&gt; Warning: replacing previous import 'rgeos::setdiff' by 'dplyr::setdiff' when</span>
<span class="co">#&gt; loading 'HierDmap'</span></code></pre></div>
</div>
</div>
<div id="simulated-hierarchical-island-model" class="section level1">
<h1 class="hasAnchor">
<a href="#simulated-hierarchical-island-model" class="anchor"></a>Simulated hierarchical island model</h1>
<p>We simulated microsatellite genotypes using fastsimcoal2 (Excoffier, L.&amp; Foll, M, 2011). There are 16 populations with a total of 320 individuals nested in 4 regions. Each region has 7, 4, 2, 3 populations respectively. Therefore, we first calculate the genetic diversity and decompose it into three spatial levels: ecosystem, region, and local population level.</p>
<p>Calculating the genetic diversity can be achieved by “HierDq” function. The hierarchical structure is easilyinputted by specifying the arguments, " <em>nreg = 4,r=c(7,4,2,3)</em> ".</p>
<p>The commonly used diversity profile includes three diversity orders.q=0, indicates the richness, q=1, indicates Shannon diversity (exponential Shannon entropy), and q=2, indicates Simpson diversity (inverse Simpson index). Users can specify which diversity you want to calculate simply give the value of <em>q</em> in <em>HierDq</em> function.</p>
<div id="load-the-data-and-calculate-genetic-diversity-profiles-" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-data-and-calculate-genetic-diversity-profiles-" class="anchor"></a>Load the data and calculate genetic diversity profiles.</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/xinghuq/HierDmap">HierDmap</a></span><span class="op">)</span>

<span class="va">f</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">'extdata'</span>,package<span class="op">=</span><span class="st">'HierDpart'</span><span class="op">)</span>
<span class="va">infile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="va">f</span>, <span class="st">"Island.gen"</span><span class="op">)</span>
<span class="va">file</span> <span class="op">=</span>  <span class="fu">diveRsity</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/diveRsity/man/readGenepop.html">readGenepop</a></span><span class="op">(</span><span class="va">infile</span>, gp<span class="op">=</span><span class="fl">2</span>, bootstrap <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>

<span class="co">### calculate diversity of order q=1, number of regions, 4, each region has 7,4,2,3 populations.</span>
<span class="va">HierDisland</span><span class="op">=</span><span class="fu"><a href="../reference/HierDq.html">HierDq</a></span><span class="op">(</span><span class="va">infile</span>,q<span class="op">=</span><span class="fl">1</span>,ncode<span class="op">=</span><span class="fl">3</span>,nreg <span class="op">=</span> <span class="fl">4</span>,r<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">7</span>,<span class="fl">4</span>,<span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>By now, using HierDq, we have calculated the genetic diversity at three spatial levels. The genetic diversity is calculated by locus. The overall diversity for each population is the median value of diversity across all loci.</p>
</div>
<div id="plot-the-hierarchy-of-diversities-from-populations-to-ecosystem" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-hierarchy-of-diversities-from-populations-to-ecosystem" class="anchor"></a>Plot the hierarchy of diversities from populations to ecosystem</h2>
<p>Now, we plot the genetic diversity in the hierarchy. Circles indicate the populations or regions. The size of the circle represents the diversity values.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">###Structure the diversity values</span>
<span class="va">hie_mapdata</span><span class="op">=</span><span class="fu"><a href="../reference/HierDstr.html">HierDstr</a></span><span class="op">(</span><span class="va">HierDisland</span><span class="op">)</span>

<span class="co">### Plot the diversity </span>
<span class="fu"><a href="../reference/HierDplot.html">HierDplot</a></span><span class="op">(</span><span class="va">hie_mapdata</span>,layout <span class="op">=</span> <span class="st">'circlepack'</span><span class="op">)</span>
<span class="co">#&gt; Non-leaf weights ignored</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig1-1.png" width="816" style="display: block; margin: auto;"></p>
<p>Fig. 1. The hierarchy of diversity (q=1) values. The size of the circle represents the diversity value.</p>
</div>
<div id="human-genome-diversity-project-hgdp-data" class="section level2">
<h2 class="hasAnchor">
<a href="#human-genome-diversity-project-hgdp-data" class="anchor"></a>Human Genome Diversity Project (HGDP) data</h2>
<p>Now, we use the Human Genome Diversity Project (HGDP) data (Cann HM,et al. 2002) to show the spatial genetic diversity distributions across geography. The dataset consists of 1350 individuals from 79 populations genotyped at 678 microsatellite loci (Cann HM et al 2002). We first calculate the genetic diversity at each locus, and then estimate the diversity at three levels: globle, continent, subcontinent, populations (within country). Finally, we plot the human genetic diversity distribution and genetic-diversity-density across geographic locations.</p>
</div>
<div id="load-the-data-and-calculate-human-genetic-diversity-profiles-" class="section level2">
<h2 class="hasAnchor">
<a href="#load-the-data-and-calculate-human-genetic-diversity-profiles-" class="anchor"></a>Load the data and calculate human genetic diversity profiles.</h2>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## load dataset, genlind format</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">eHGDP</span><span class="op">)</span>

<span class="co">### formating data, restructing the hierarchical data structure </span>
<span class="fu"><a href="https://rdrr.io/r/base/levels.html">levels</a></span><span class="op">(</span><span class="va">eHGDP</span><span class="op">$</span><span class="va">pop</span><span class="op">)</span><span class="op">=</span><span class="va">eHGDP</span><span class="op">$</span><span class="va">other</span><span class="op">$</span><span class="va">popInfo</span><span class="op">$</span><span class="va">Population</span>
<span class="co">###struct a table with individual region/pop labels</span>
<span class="va">popstr</span><span class="op">=</span> <span class="va">eHGDP</span><span class="op">$</span><span class="va">other</span><span class="op">$</span><span class="va">popInfo</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">eHGDP</span><span class="op">$</span><span class="va">pop</span>,<span class="va">eHGDP</span><span class="op">$</span><span class="va">other</span><span class="op">$</span><span class="va">popInfo</span><span class="op">$</span><span class="va">Population</span><span class="op">)</span>,<span class="op">]</span>
<span class="va">popstr</span><span class="op">$</span><span class="va">Population</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">popstr</span><span class="op">$</span><span class="va">Population</span>,levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">popstr</span><span class="op">$</span><span class="va">Population</span><span class="op">)</span><span class="op">)</span>
<span class="va">popstr</span><span class="op">$</span><span class="va">Region</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">popstr</span><span class="op">$</span><span class="va">Region</span>,levels<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">popstr</span><span class="op">$</span><span class="va">Region</span><span class="op">)</span><span class="op">)</span>
<span class="co">#library(poppr)</span>
<span class="co">#eHGDP1=missingno(eHGDP, type = "mean",  quiet = FALSE, freq = FALSE)</span>

<span class="va">eHGDP</span><span class="op">$</span><span class="va">tab</span><span class="op">=</span><span class="op">(</span><span class="fu">adegenet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/adegenet/man/tab.html">tab</a></span><span class="op">(</span><span class="va">eHGDP</span>, freq <span class="op">=</span> <span class="cn">FALSE</span>, NA.method <span class="op">=</span> <span class="st">"mean"</span><span class="op">)</span><span class="op">)</span>

<span class="co">#HiereHGDPq0=HierDgenind(eHGDP,q=0,pop_region = popstr$Region,pop = popstr$Population)</span>
<span class="co">### we calculate the diversity of order q=1</span>
<span class="va">HiereHGDPq1</span><span class="op">=</span><span class="fu"><a href="../reference/HierDgenind.html">HierDgenind</a></span><span class="op">(</span><span class="va">eHGDP</span>,q<span class="op">=</span><span class="fl">1</span>,pop_region <span class="op">=</span> <span class="va">popstr</span><span class="op">$</span><span class="va">Region</span>,pop <span class="op">=</span> <span class="va">popstr</span><span class="op">$</span><span class="va">Population</span><span class="op">)</span>
<span class="co">#HiereHGDPq2=HierDgenind(eHGDP,q=2,pop_region = popstr$Region,pop = popstr$Population)</span></code></pre></div>
</div>
<div id="plotting-the-human-genetic-diversity-in-the-hierarchy-from-countries-to-continents-" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-human-genetic-diversity-in-the-hierarchy-from-countries-to-continents-" class="anchor"></a>Plotting the human genetic diversity in the hierarchy, from countries to continents.</h2>
<p>We first format the diversity values obtained from “HierDgenind” function using <em>HierDstrp</em> function. We then plot the the diversity hierarchy using <em>HierDplot</em> function.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">###Structure the diversity values</span>
<span class="va">a</span><span class="op">=</span><span class="fu"><a href="../reference/HierDstrp.html">HierDstrp</a></span><span class="op">(</span><span class="va">HiereHGDPq1</span><span class="op">)</span>

<span class="co">### Plot the diversity </span>
<span class="fu"><a href="../reference/HierDplot.html">HierDplot</a></span><span class="op">(</span><span class="va">a</span>,layout <span class="op">=</span> <span class="st">'circlepack'</span><span class="op">)</span><span class="op">+</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_brewer.html">scale_fill_distiller</a></span><span class="op">(</span>palette <span class="op">=</span> <span class="st">"RdPu"</span><span class="op">)</span>
<span class="co">#&gt; Non-leaf weights ignored</span>
<span class="co">#&gt; Scale for 'fill' is already present. Adding another scale for 'fill', which</span>
<span class="co">#&gt; will replace the existing scale.</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig2-1.png" width="816" style="display: block; margin: auto;"></p>
<p>Fig. 2. The genetic diversity (q=1) of HGDP data in the hierarchical structure. The size of the circle represents the diversity size.</p>
</div>
<div id="plotting-the-interactive-plot-for-genetic-diversities-in-the-hierarchical-structure-from-countries-to-continents-" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-interactive-plot-for-genetic-diversities-in-the-hierarchical-structure-from-countries-to-continents-" class="anchor"></a>Plotting the interactive plot for genetic diversities in the hierarchical structure, from countries to continents.</h2>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="co">###Structure the diversity values</span>
<span class="va">a</span><span class="op">=</span><span class="fu"><a href="../reference/HierDstrp.html">HierDstrp</a></span><span class="op">(</span><span class="va">HiereHGDPq1</span><span class="op">)</span>

<span class="co">### Plot the diversity </span>

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">)</span>

<span class="fu"><a href="../reference/PlothieD.html">PlothieD</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">[</span>,<span class="op">-</span><span class="fl">1</span><span class="op">]</span>,size <span class="op">=</span> <span class="st">"Dq1"</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-794e0608c0aa9e7ca813" style="width:816px;height:768px;" class="circlepackeR html-widget"></div>
<script type="application/json" data-for="htmlwidget-794e0608c0aa9e7ca813">{"x":{"data":{"name":"world","children":[{"name":"EUROPE","children":[{"name":"EUROPE","children":[{"name":"Orcadian","Dq0":60,"Dq1":72.9777,"Dq2":74.8918,"Label":1,"Latitude":59,"Longitude":-3,"Population":"Orcadian","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Adygei","Dq0":80,"Dq1":72.075,"Dq2":73.4071,"Label":2,"Latitude":44,"Longitude":39,"Population":"Adygei","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Russian","Dq0":80,"Dq1":73.0993,"Dq2":75.6981,"Label":3,"Latitude":61,"Longitude":40,"Population":"Russian","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Basque","Dq0":80,"Dq1":71.3495,"Dq2":71.736,"Label":4,"Latitude":43,"Longitude":0,"Population":"Basque","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"French","Dq0":80,"Dq1":78.3977,"Dq2":79.1279,"Label":5,"Latitude":46,"Longitude":2,"Population":"French","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Italian","Dq0":60,"Dq1":72.8427,"Dq2":74.3252,"Label":6,"Latitude":46,"Longitude":10,"Population":"Italian","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Sardinian","Dq0":80,"Dq1":71.5717,"Dq2":73.6766,"Label":7,"Latitude":40,"Longitude":9,"Population":"Sardinian","Region":"EUROPE","Sub_region":"EUROPE"},{"name":"Tuscan","Dq0":60,"Dq1":79.8098,"Dq2":74.9902,"Label":8,"Latitude":43,"Longitude":11,"Population":"Tuscan","Region":"EUROPE","Sub_region":"EUROPE"}]}]},{"name":"ASIA","children":[{"name":"MIDDLE_EAST","children":[{"name":"Mozabite","Dq0":80,"Dq1":81.8065,"Dq2":84.4563,"Label":9,"Latitude":32,"Longitude":3,"Population":"Mozabite","Region":"ASIA","Sub_region":"MIDDLE_EAST"},{"name":"Bedouin","Dq0":100,"Dq1":81.5029,"Dq2":81.1424,"Label":10,"Latitude":31,"Longitude":35,"Population":"Bedouin","Region":"ASIA","Sub_region":"MIDDLE_EAST"},{"name":"Druze","Dq0":80,"Dq1":74.3091,"Dq2":74.6684,"Label":11,"Latitude":32,"Longitude":35,"Population":"Druze","Region":"ASIA","Sub_region":"MIDDLE_EAST"},{"name":"Palestinian","Dq0":100,"Dq1":79.6043,"Dq2":80.4976,"Label":12,"Latitude":32,"Longitude":35,"Population":"Palestinian","Region":"ASIA","Sub_region":"MIDDLE_EAST"}]},{"name":"CENTRAL_SOUTH_ASIA","children":[{"name":"Balochi","Dq0":80,"Dq1":77.7845,"Dq2":78.1815,"Label":13,"Latitude":30.5,"Longitude":66.5,"Population":"Balochi","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Brahui","Dq0":80,"Dq1":78.2852,"Dq2":77.1679,"Label":14,"Latitude":30.5,"Longitude":66.5,"Population":"Brahui","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Burusho","Dq0":80,"Dq1":74.2407,"Dq2":76.8264,"Label":15,"Latitude":36.5,"Longitude":74,"Population":"Burusho","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Hazara","Dq0":80,"Dq1":77.3454,"Dq2":77.3307,"Label":16,"Latitude":33.5,"Longitude":70,"Population":"Hazara","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Kalash","Dq0":60,"Dq1":59.6621,"Dq2":62.9586,"Label":17,"Latitude":36,"Longitude":71.5,"Population":"Kalash","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Makrani","Dq0":80,"Dq1":81.3244,"Dq2":82.0545,"Label":18,"Latitude":26,"Longitude":64,"Population":"Makrani","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Pathan","Dq0":80,"Dq1":77.8835,"Dq2":79.7864,"Label":19,"Latitude":33.5,"Longitude":70.5,"Population":"Pathan","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"},{"name":"Sindhi","Dq0":80,"Dq1":78.7018,"Dq2":79.889,"Label":20,"Latitude":25.5,"Longitude":69,"Population":"Sindhi","Region":"ASIA","Sub_region":"CENTRAL_SOUTH_ASIA"}]},{"name":"EAST_ASIA","children":[{"name":"Han","Dq0":80,"Dq1":68.6671,"Dq2":69.8099,"Label":35,"Latitude":32.5,"Longitude":114,"Population":"Han","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Han-NChina","Dq0":60,"Dq1":70.1497,"Dq2":66.0932,"Label":36,"Latitude":39,"Longitude":114,"Population":"Han-NChina","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Dai","Dq0":60,"Dq1":66.3605,"Dq2":65.0863,"Label":37,"Latitude":21,"Longitude":100,"Population":"Dai","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Daur","Dq0":60,"Dq1":66.2934,"Dq2":65.7221,"Label":38,"Latitude":48.5,"Longitude":124,"Population":"Daur","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Hezhen","Dq0":60,"Dq1":60.3431,"Dq2":62.9657,"Label":39,"Latitude":47.5,"Longitude":133.5,"Population":"Hezhen","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Lahu","Dq0":40,"Dq1":54.1701,"Dq2":57.8749,"Label":40,"Latitude":22,"Longitude":100,"Population":"Lahu","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Miao","Dq0":60,"Dq1":64.4773,"Dq2":61.8797,"Label":41,"Latitude":28,"Longitude":109,"Population":"Miao","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Oroqen","Dq0":60,"Dq1":60.3431,"Dq2":62.4938,"Label":42,"Latitude":50.5,"Longitude":126.5,"Population":"Oroqen","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"She","Dq0":60,"Dq1":58.8093,"Dq2":59.3605,"Label":43,"Latitude":27,"Longitude":119,"Population":"She","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Tujia","Dq0":60,"Dq1":60.9974,"Dq2":64.0554,"Label":44,"Latitude":29,"Longitude":109,"Population":"Tujia","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Tu","Dq0":60,"Dq1":62.8566,"Dq2":65.0863,"Label":45,"Latitude":36,"Longitude":101,"Population":"Tu","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Xibo","Dq0":60,"Dq1":68.2153,"Dq2":66.299,"Label":46,"Latitude":43.5,"Longitude":81.5,"Population":"Xibo","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Yi","Dq0":60,"Dq1":68.4122,"Dq2":65.9794,"Label":47,"Latitude":28,"Longitude":103,"Population":"Yi","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Mongola","Dq0":60,"Dq1":70.5951,"Dq2":68.5093,"Label":48,"Latitude":48.5,"Longitude":119,"Population":"Mongola","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Naxi","Dq0":60,"Dq1":58.2177,"Dq2":63.3539,"Label":49,"Latitude":26,"Longitude":100,"Population":"Naxi","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Uygur","Dq0":80,"Dq1":79.2036,"Dq2":82.1957,"Label":50,"Latitude":44,"Longitude":81,"Population":"Uygur","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Cambodian","Dq0":60,"Dq1":71.7312,"Dq2":69.984,"Label":51,"Latitude":12,"Longitude":105,"Population":"Cambodian","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Japanese","Dq0":80,"Dq1":66.6105,"Dq2":68.426,"Label":52,"Latitude":38,"Longitude":138,"Population":"Japanese","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"Yakut","Dq0":80,"Dq1":64.7114,"Dq2":64.4133,"Label":53,"Latitude":63,"Longitude":129.5,"Population":"Yakut","Region":"ASIA","Sub_region":"EAST_ASIA"},{"name":"TundraNentsi","Dq0":60,"Dq1":68.5534,"Dq2":70.0026,"Label":55,"Latitude":67,"Longitude":65,"Population":"TundraNentsi","Region":"ASIA","Sub_region":"EAST_ASIA"}]}]},{"name":"OCEANIA","children":[{"name":"OCEANIA","children":[{"name":"Melanesian","Dq0":60,"Dq1":50.9671,"Dq2":53.9334,"Label":21,"Latitude":-6,"Longitude":155,"Population":"Melanesian","Region":"OCEANIA","Sub_region":"OCEANIA"},{"name":"Papuan","Dq0":60,"Dq1":58.3449,"Dq2":58.1511,"Label":22,"Latitude":-4,"Longitude":143,"Population":"Papuan","Region":"OCEANIA","Sub_region":"OCEANIA"}]}]},{"name":"AMERICA","children":[{"name":"AMERICA","children":[{"name":"Colombian","Dq0":20,"Dq1":32.0414,"Dq2":34.1929,"Label":23,"Latitude":3,"Longitude":-68,"Population":"Colombian","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Karitiana","Dq0":20,"Dq1":17.4895,"Dq2":19.3918,"Label":24,"Latitude":-10,"Longitude":-63,"Population":"Karitiana","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Maya","Dq0":60,"Dq1":52.9744,"Dq2":51.5258,"Label":25,"Latitude":19,"Longitude":-91,"Population":"Maya","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Pima","Dq0":20,"Dq1":29.8795,"Dq2":32.1301,"Label":26,"Latitude":29,"Longitude":-108,"Population":"Pima","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Surui","Dq0":0,"Dq1":1.1398,"Dq2":0.8579,"Label":54,"Latitude":-11,"Longitude":-62,"Population":"Surui","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Chipewyan","Dq0":60,"Dq1":50.9206,"Dq2":50.4223,"Label":56,"Latitude":59.55,"Longitude":-107.3,"Population":"Chipewyan","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Cree","Dq0":60,"Dq1":62.9496,"Dq2":63.6325,"Label":57,"Latitude":50.33,"Longitude":-102.5,"Population":"Cree","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Ojibwa","Dq0":60,"Dq1":60.3744,"Dq2":59.1389,"Label":58,"Latitude":46.5,"Longitude":-81,"Population":"Ojibwa","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Kaqchikel","Dq0":40,"Dq1":46.7722,"Dq2":47.6523,"Label":59,"Latitude":15,"Longitude":-91,"Population":"Kaqchikel","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Mixtec","Dq0":40,"Dq1":42.167,"Dq2":43.2103,"Label":60,"Latitude":17,"Longitude":-97,"Population":"Mixtec","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Mixe","Dq0":40,"Dq1":40.8852,"Dq2":39.9799,"Label":61,"Latitude":17,"Longitude":-96,"Population":"Mixe","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Zapotec","Dq0":60,"Dq1":54.0322,"Dq2":53.2181,"Label":62,"Latitude":16,"Longitude":-97,"Population":"Zapotec","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Guaymi","Dq0":20,"Dq1":24.4921,"Dq2":26.929,"Label":63,"Latitude":8.5,"Longitude":-82,"Population":"Guaymi","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Cabecar","Dq0":40,"Dq1":37.3688,"Dq2":36.3557,"Label":64,"Latitude":9.5,"Longitude":-84,"Population":"Cabecar","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Aymara","Dq0":60,"Dq1":48.6039,"Dq2":46.2522,"Label":65,"Latitude":-22,"Longitude":-70,"Population":"Aymara","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Huilliche","Dq0":60,"Dq1":54.2737,"Dq2":50.6833,"Label":66,"Latitude":-41,"Longitude":-73,"Population":"Huilliche","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Guarani","Dq0":40,"Dq1":39.6218,"Dq2":39.0457,"Label":67,"Latitude":-23,"Longitude":-54,"Population":"Guarani","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Ache","Dq0":0,"Dq1":0,"Dq2":0,"Label":68,"Latitude":-24,"Longitude":-56,"Population":"Ache","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Kaingang","Dq0":20,"Dq1":32.0414,"Dq2":30.9685,"Label":69,"Latitude":-24,"Longitude":-52.5,"Population":"Kaingang","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Quechua","Dq0":60,"Dq1":52.9037,"Dq2":51.4195,"Label":70,"Latitude":-14,"Longitude":-74,"Population":"Quechua","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Kogi","Dq0":0,"Dq1":14.8729,"Dq2":17.0707,"Label":71,"Latitude":11,"Longitude":-74,"Population":"Kogi","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Zenu","Dq0":40,"Dq1":39.923,"Dq2":39.688,"Label":72,"Latitude":9,"Longitude":-75,"Population":"Zenu","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Inga","Dq0":40,"Dq1":40.0373,"Dq2":37.3067,"Label":73,"Latitude":1,"Longitude":-77,"Population":"Inga","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Wayuu","Dq0":60,"Dq1":52.0582,"Dq2":50.616,"Label":74,"Latitude":11,"Longitude":-73,"Population":"Wayuu","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"TicunaArara","Dq0":20,"Dq1":21.6079,"Dq2":23.2357,"Label":75,"Latitude":-4,"Longitude":-70,"Population":"TicunaArara","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"TicunaTarapaca","Dq0":20,"Dq1":20.632,"Dq2":24.7781,"Label":76,"Latitude":-4,"Longitude":-70,"Population":"TicunaTarapaca","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Embera","Dq0":40,"Dq1":36.892,"Dq2":34.1929,"Label":77,"Latitude":7,"Longitude":-76,"Population":"Embera","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Waunana","Dq0":40,"Dq1":39.3202,"Dq2":38.5016,"Label":78,"Latitude":5,"Longitude":-77,"Population":"Waunana","Region":"AMERICA","Sub_region":"AMERICA"},{"name":"Arhuaco","Dq0":20,"Dq1":32.9398,"Dq2":36.0659,"Label":79,"Latitude":11,"Longitude":-73.8,"Population":"Arhuaco","Region":"AMERICA","Sub_region":"AMERICA"}]}]},{"name":"AFRICA","children":[{"name":"AFRICA","children":[{"name":"BantuKenya","Dq0":80,"Dq1":97.2916,"Dq2":95.6586,"Label":27,"Latitude":-3,"Longitude":37,"Population":"BantuKenya","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"BantuSouthWest","Dq0":40,"Dq1":56.3057,"Dq2":87.574,"Label":28,"Latitude":-21,"Longitude":18.7,"Population":"BantuSouthWest","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"BantuSouthEast","Dq0":60,"Dq1":76.8011,"Dq2":80.8657,"Label":29,"Latitude":-28.4,"Longitude":27.6,"Population":"BantuSouthEast","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"Mandenka","Dq0":80,"Dq1":91.6637,"Dq2":93.004,"Label":30,"Latitude":12,"Longitude":-12,"Population":"Mandenka","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"Yoruba","Dq0":100,"Dq1":100,"Dq2":97.9963,"Label":31,"Latitude":8,"Longitude":5,"Population":"Yoruba","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"BiakaPygmy","Dq0":100,"Dq1":93.5881,"Dq2":100,"Label":32,"Latitude":4,"Longitude":17,"Population":"BiakaPygmy","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"MbutiPygmy","Dq0":80,"Dq1":89.4032,"Dq2":91.9892,"Label":33,"Latitude":1,"Longitude":29,"Population":"MbutiPygmy","Region":"AFRICA","Sub_region":"AFRICA"},{"name":"San","Dq0":60,"Dq1":96.9708,"Dq2":86.2736,"Label":34,"Latitude":-21,"Longitude":20,"Population":"San","Region":"AFRICA","Sub_region":"AFRICA"}]}]}]},"options":{"size":"Dq1","color_min":"hsl(56,80%,80%)","color_max":"hsl(341,30%,40%)"}},"evals":[],"jsHooks":[]}</script><p>Fig.3. The interactive plot of genetic diversity (q=1) across continents for HGDP data. The size of the circle represents the diversity value.</p>
</div>
<div id="human-genetic-diversity-distribution-and-genetic-diversity-density-across-geographic-location" class="section level2">
<h2 class="hasAnchor">
<a href="#human-genetic-diversity-distribution-and-genetic-diversity-density-across-geographic-location" class="anchor"></a>Human genetic diversity distribution and genetic-diversity-density across geographic location</h2>
<p>We plot the human genetic diversity distribution and genetic-diversity-density across geographic location based on the genetic diversity values calculated above. There are four methods for plotting the genetic diversity distributions. I demonstrate the four methods below. The dataset used to plot the geographic distribution should be formatted in a dataframe including the diversity values and the sample locations (longitude and latitude).</p>
</div>
<div id="plotting-the-geo-diversity-map-method-1-" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-geo-diversity-map-method-1-" class="anchor"></a>Plotting the geo-diversity map method 1.</h2>
<p>This method uses r graphics and rgeos to locate the diversity values to the map. Sometimes, the diversity values have the same magnitude, so we can not distinguish the size of the diversity. But we can scale the values between 0-10 and then plot the relative size. But this depends on your datasets.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">##load data</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">)</span>

<span class="fu"><a href="../reference/rDivgeomap.html">rDivgeomap</a></span><span class="op">(</span>x<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Longitude</span>,y<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Latitude</span>,size<span class="op">=</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Dq1</span><span class="op">)</span><span class="op">*</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig4-1.png" width="816" style="display: block; margin: auto;"> Fig. 4. Geo-diversity map of HGDP populations.</p>
</div>
<div id="plotting-the-geo-diversity-map-method-2-" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-geo-diversity-map-method-2-" class="anchor"></a>Plotting the geo-diversity map method 2.</h2>
<p>This method uses ggplot2.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## </span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">)</span>

<span class="fu"><a href="../reference/ggDivgeomap.html">ggDivgeomap</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">Dprofile</span>,x<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Longitude</span>, y<span class="op">=</span> <span class="va">Dprofile</span><span class="op">$</span><span class="va">Latitude</span>,scale <span class="op">=</span> <span class="st">"medium"</span>, returnclass <span class="op">=</span> <span class="st">"sf"</span>,size<span class="op">=</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Dq1</span><span class="op">)</span><span class="op">*</span><span class="fl">10</span>,title<span class="op">=</span><span class="st">"Human Population Genetic Diversity Map"</span>,subtitle<span class="op">=</span><span class="op">(</span><span class="st">"Scaled relative genetic diversity of HGDP populations"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig5-1.png" width="816" style="display: block; margin: auto;"> Fig. 5. Geo-diversity map of HGDP populations.</p>
</div>
<div id="plotting-the-geo-diversity-and-diversity-density-map-method-3-" class="section level2">
<h2 class="hasAnchor">
<a href="#plotting-the-geo-diversity-and-diversity-density-map-method-3-" class="anchor"></a>Plotting the geo-diversity and diversity-density map method 3.</h2>
<p>This method uses ggplot2 to plot the geographic map. But this function also plots the density of the diversity values. We scale the values between 0-10 and then plot the relative size. The genetic diversity is estimated based on the relative values of the genetic diversity distributed across the space.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">)</span>

 <span class="fu"><a href="../reference/dDivgeomap.html">dDivgeomap</a></span><span class="op">(</span><span class="va">Dprofile</span>,x<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Longitude</span>, y<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Latitude</span>,label<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Population</span>,border_colour<span class="op">=</span><span class="cn">NA</span>, border_fill<span class="op">=</span><span class="st">"antiquewhite"</span>,density_fill <span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Dq1</span> , density_alpha <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="fl">.2</span><span class="op">)</span>,density_size <span class="op">=</span> <span class="fl">1</span>, bins <span class="op">=</span> <span class="fl">5</span>, geom <span class="op">=</span> <span class="st">"polygon"</span>,point_color<span class="op">=</span><span class="st">"red"</span>, pointalpha <span class="op">=</span> <span class="fl">.2</span>, pointsize<span class="op">=</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Dq1</span><span class="op">)</span><span class="op">*</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig6-1.png" width="816" style="display: block; margin: auto;"></p>
<p>Fig. 6. Geo-diversity and diversity density map of HGDP populations.</p>
</div>
<div id="plot-the-geo-diversity-map-method-4-" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-the-geo-diversity-map-method-4-" class="anchor"></a>Plot the geo-diversity map method 4.</h2>
<p>This method uses ggmap and ggplpt2 to plot the geographic map and locate the diversity values on the map.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R">
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">)</span>
<span class="fu"><a href="../reference/dgDivgeomap.html">dgDivgeomap</a></span><span class="op">(</span><span class="va">Dprofile</span>,x<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Longitude</span>, y<span class="op">=</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Latitude</span>,lon <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">160</span>, <span class="fl">160</span><span class="op">)</span>, lat <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">60</span>, <span class="fl">70</span><span class="op">)</span>,  mapsource <span class="op">=</span> <span class="st">"osm"</span>, mapcolor <span class="op">=</span> <span class="st">"color"</span>, maptype<span class="op">=</span><span class="st">"satellite"</span>,
                     point_color<span class="op">=</span><span class="st">"red"</span>, pointalpha <span class="op">=</span> <span class="fl">.2</span>, pointsize<span class="op">=</span><span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span><span class="va">Dprofile</span><span class="op">$</span><span class="va">Dq1</span><span class="op">)</span><span class="op">*</span><span class="fl">10</span><span class="op">)</span></code></pre></div>
<p><img src="introduction_files/figure-html/fig7-1.png" width="816" style="display: block; margin: auto;"> Fig. 7. Geo-diversity and diversity density map of HGDP populations.</p>
<p>The above vignette summarizes and demonstrates the main functions of “HierDmap” package. Users can apply this package and adopt the methods for visualizing different values and hierarchies. This package will be much useful for metacommunity data, metapopulation data, and for geographic distribution/density of the target metric.</p>
</div>
<div id="references" class="section level2">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<p>Chao, A., Chiu, C. H., &amp; Jost, L. (2014). Unifying species diversity, phylogenetic diversity, functional diversity, and related similarity and differentiation measures through Hill numbers. Annual review of ecology, evolution, and systematics, 45, 297-324.</p>
<p>Gaggiotti, O. E., Chao, A., Peres‐Neto, P., Chiu, C. H., Edwards, C., Fortin, M. J., … &amp; Selkoe, K. A. (2018). Diversity from genes to ecosystems: A unifying framework to study variation across biological metrics and scales. Evolutionary Applications, 11(7), 1176-1193.</p>
<p>Jost, L., Archer, F., Flanagan, S., Gaggiotti, O., Hoban, S., &amp; Latch, E. (2018). Differentiation measures for conservation genetics. Evolutionary Applications, 11(7), 1139-1148.</p>
<p>Sherwin, W. B. (2018). Entropy, or information, unifies ecology and evolution and beyond. Entropy, 20(10), 727.</p>
<p>Cann HM, de Toma C, Cazes L, Legrand MF, Morel V, et al. (2002) A human genome diversity cell line panel. Science 296: 261-262.</p>
<p>Excoffier, L. and Foll, M (2011) fastsimcoal: a continuous-time coalescent simulator of genomic diversity under arbitrarily complex evolutionary scenarios Bioinformatics 27: 1332-1334.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Xinghu Qin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
